Add documentation for core Docker concepts and guides (#19691)
docker/docs: content/guides/docker-concepts/the-basics/what-is-docker-compose.md

---
title: What is Docker Compose?
keywords: concepts, build, images, container, docker desktop
description: What is Docker Compose?
---

{{< youtube-embed xhcUIK4fGtY >}}

## Explanation

If you've been following the guides so far, you've been working with single container applications. But, now you're wanting to do something more complicated - run databases, message queues, caches, or a variety of other services. Do you install everything in a single container? Run multiple containers? If you run multiple, how do you connect them all together?

> One container best practice is that each container should do one thing and do it well. While there are exceptions to this rule, avoid the tendency to have one container do multiple things.

Reviewer: I tripped up reading this the first time. Maybe reword to:

```suggestion
One best practice for containers is that each container should do one thing and do it well. While there are exceptions to this rule, avoid the tendency to have one container do multiple things.
```

    ✔ Network todo-list-app_default    Removed        0.1s
    ```

    > **Volume persistence**
    >
    > By default, volumes _aren't_ automatically removed when you tear down a Compose stack. The idea is that you might want the data back if you start the stack again.
    >
    > If you do want to remove the volumes, add the `--volumes` flag when running the `docker compose down` command:
    >
    > ```console
    > docker compose down --volumes
    > ```
    { .tip }

2. Alternatively, you can use the Docker Desktop GUI to remove the containers by selecting the application stack and selecting the **Delete** button.

    ![A screenshot of the Docker Desktop GUI showing the containers view with an arrow pointing to the "Delete" button](images/todo-list-delete.webp?w=930&h=400)

    > **Using the GUI for Compose stacks**
    >
    > Note that if you remove the containers for a Compose app in the GUI, it's removing only the containers. You'll have to manually remove the network and volumes if you want to do so.
    { .important }

In this walkthrough, you learned how to use Docker Compose to start and stop a multi-container application.

## Additional resources

> This concept page purposely didn't do a deep-dive on Compose and how to write a Compose file, but focused on introducing you to what Compose is.

Reviewer: Maybe reword to have a CTA.

```suggestion
This page was a brief introduction Compose. In the following resources, you can dive deeper into Compose and how to write a Compose files.
```

Author: Adjusted the wording.
